### DISCUSSION TITLE
Trouble with exe created for spacy 3.1.1

### DISCUSSION THREAD

**Discussion by erotavlas** [pyinstaller/pyinstaller#6106] [Help]
I'm having trouble creating a stable exe for spacy 3.1.1.  When I load a model (also trained in spacy 3.1.1)  I get the following error
Does anyone have any luck creating exe for spacy version 3.1.1, if so any ideas what I'm missing?

I'm on Windows 10 machine, and Python 3.8.5

> Error 3:<class 'catalogue.RegistryError'>[E893] Could not find function 'spacy.MultiHashEmbed.v1' in function registry 'architectures'. If you're using a custom function, make sure the code is available. If the function is provided by a third-party package, e.g. spacy-transformers, make sure the package is installed in your environment.
> 
> Available names: spacy.CharacterEmbed.v2, spacy.EntityLinker.v1, spacy.HashEmbedCNN.v2, spacy.MaxoutWindowEncoder.v2, spacy.MishWindowEncoder.v2, spacy.MultiHashEmbed.v2, spacy.PretrainCharacters.v1, spacy.PretrainVectors.v1, spacy.SpanCategorizer.v1, spacy.Tagger.v1, spacy.TextCatBOW.v2, spacy.TextCatCNN.v2, spacy.TextCatEnsemble.v2, spacy.TextCatLowData.v1, spacy.Tok2Vec.v2, spacy.Tok2VecListener.v1, spacy.TorchBiLSTMEncoder.v1, spacy.TransitionBasedParser.v2

My spec file looks like

```
# -*- mode: python ; coding: utf-8 -*-

import PyInstaller
from PyInstaller.utils.hooks import collect_submodules
from PyInstaller.utils.hooks import copy_metadata

copy_metadata('spacy_legacy')

block_cipher = None

datas = [
    ('G:\TESTING\patterns.json', 'patterns')
]

datas.extend(PyInstaller.utils.hooks.collect_data_files('thinc'))
datas.extend(PyInstaller.utils.hooks.collect_data_files('spacy'))

all_hidden_imports = [
             'spacy.lexeme',
             'spacy.lang.en',
             'spacy.parts_of_speech',
             'spacy.tokens._dict_proxies',
             'spacy.lang.norm_exceptions',
             'spacy.lang.lex_attrs',
             'spacy.pipeline.transition_parser',
             'spacy.pipeline._parser_internals.stateclass',
             'spacy.pipeline._parser_internals._beam_utils',
             'spacy.pipeline._parser_internals.transition_system',
             'spacy.pipeline._parser_internals.arc_eager',
             'spacy.pipeline._parser_internals.ner',
             'srsly.msgpack.util', 
             'cymem', 
             'cymem.cymem', 
             'preshed.maps',
             'thinc.backends.linalg',
             'thinc.extra.search',
             'blis',
             'blis.py',
             ] + collect_submodules('spacy_legacy')

a = Analysis(['ner_ai.py'],
             pathex=['D:\\SOURCECODE\\PYTHON\\pythonInterop\\pyInstaller'],
             binaries=[],
             datas=datas,
             hiddenimports=all_hidden_imports,
             hookspath=[],
             runtime_hooks=[],
             excludes=['torch'],
             win_no_prefer_redirects=False,
             win_private_assemblies=False,
             cipher=block_cipher,
             noarchive=False)
pyz = PYZ(a.pure, a.zipped_data,
             cipher=block_cipher)
exe = EXE(pyz,
          a.scripts,
          [],
          exclude_binaries=True,
          name='ner',
          debug=False,
          bootloader_ignore_signals=False,
          strip=False,
          upx=True,
          console=True )
coll = COLLECT(exe,
               a.binaries,
               a.zipfiles,
               a.datas,
               strip=False,
               upx=True,
               upx_exclude=[],
               name='ner')

```

**Comment by erotavlas** [comment]
Fixed by upgrading config to version 3.1 then changing MultiHashEmbed from version 1 to V2 in the config file, then retrained model.

**Reply by DarkSoliditi** [reply]
What was the implementation like? Did you you have to create a hidden imports .py file or did you just copy and paste it into the spec file?

**Reply by erotavlas** [reply]
I don't understand your question.  Are you having trouble creating the exe?
